src/index_runner/es_indexers/narrative.py

```diff
@@ -24,16 +26,56 @@ def index_narrative(obj_data, ws_info, obj_data_v1):
         - created and updated dates
         - total number of cells
     """
+    # Reference for the workspace info type:
+    #    https://kbase.us/services/ws/docs/Workspace.html#typedefWorkspace.workspace_info
+    # Refer for the object info type:
+    #    https://kbase.us/services/ws/docs/Workspace.html#typedefWorkspace.object_info
     obj_info = obj_data['info']
     obj_id = obj_info[0]
-    workspace_id = obj_data['info'][6]
-    ws_id = obj_info[6]
+    obj_metadata = obj_info[-1]
+    if not obj_metadata:
+        raise RuntimeError(f"Cannot index narrative: no metadata for the narrative object. Obj info: {obj_info}")
+    [ws_id, _, owner, moddate, _, _, _, _, ws_metadata] = ws_info
+    if not ws_metadata:
+        raise RuntimeError(f"Cannot index narrative: no metadata for the workspace. WS info: {ws_info}")
+    is_temporary = _narrative_is_temporary(ws_metadata)
+    is_narratorial = _narrative_is_narratorial(ws_metadata)
+    narrative_title = obj_metadata.get('name')
+    creator = obj_data['creator']
     # Get all the types and names of objects in the narrative's workspace.
-    narrative_data_objects = indexer_utils.fetch_objects_in_workspace(ws_id)
+    narrative_data_objects = _fetch_objects_in_workspace(ws_id)
+    # Extract all the data we want to index from the notebook cells
+    raw_cells = obj_data['data'].get('cells', [])
+    index_cells = _extract_cells(raw_cells, ws_id)
+    result = {
+        '_action': 'index',
+        'doc': {
+            'narrative_title': narrative_title,
```

Comment:
New fields are: is_temporary, is_narratorial, owner, modified_at

Now that I look at it, I'm not sure why we added "modified_at" when we already have that timestamp in the generic fields (global_mappings.ws_object.timestamp). I should have caught this before.

Reply:
is the plan to change before this gets merged?